### PR TITLE
[scope] update core.demangle for 'scope' mangling

### PR DESCRIPTION
**Pull request by WalterBright**
Goes with mangling change in https://github.com/dlang/dmd/pull/6496 although 6496 does not have to be pulled first.

### CODE REVIEW

**Comment by WalterBright** [comment]
Here's what I get for the Project Tester failure:
```
GitHub pull request #1751 of commit 0b6d766c1dcf7452aa867ccedf5e77d1de4f0f7b, no merge conflicts.
Setting status of 0b6d766c1dcf7452aa867ccedf5e77d1de4f0f7b to PENDING with url https://ci.dawg.eu/job/druntime_trigger/81/ and message: 'Build started sha1 is merged.'
Using context: Project Tester
Building in workspace /var/lib/jenkins/jobs/druntime_trigger/workspace
Waiting for the completion of DPL
DPL #1110 completed. Result was FAILURE
Build step 'Trigger/call builds on other projects' marked build as failure
Setting status of 0b6d766c1dcf7452aa867ccedf5e77d1de4f0f7b to FAILURE with url https://ci.dawg.eu/job/druntime_trigger/81/ and message: 'Build finished. '
Using context: Project Tester
Finished: FAILURE
```
I don't know what to do with that.

**Comment by rainers** [comment]
>Here's what I get for the Project Tester failure:

The projects dub and vibe.d seem to always fail ATM due to added `final` to core.thread.Fiber.

File: src/core/demangle.d [hunk 1]
```diff
@@ -955,7 +960,7 @@ private struct Demangle
     {
         // FuncAttrs
         breakFuncAttrs:
-        while( 'N' == front )
+        while ('N' == front)
```
Comment (rainers) [inline src/core/demangle.d]: While the code style is rather ugly in this file, I think we should not mix different styles.

**Comment by ZombineDev** [comment]
> The projects dub and vibe.d seem to always fail ATM due to added final to core.thread.Fiber.

`Fiber.state` has been always marked as `final`: https://github.com/dlang/druntime/blob/druntime-2.052/src/core/thread.d#L3226.

**Pull request merged**